python/lance/_lib.pyx

```diff
@@ -57,6 +60,39 @@ cdef extern from "lance/arrow/writer.h" namespace "lance::arrow" nogil:
             optional[CFileWriteOptions] options)
 
 
+cdef extern from "lance/arrow/scanner.h" namespace "lance::arrow" nogil:
+    cdef cppclass LScannerBuilder "::lance::arrow::ScannerBuilder":
+        LScannerBuilder(shared_ptr[CDataset]) except +
+        void Project(const vector[string]& columns)
+        void Filter(CExpression filter)
+        void Limit(int64_t limit, int64_t offset)
+        CResult[shared_ptr[CScanner]] Finish()
+
+def BuildScanner(
+        dataset: Dataset,
+        columns: Optional[list[str]] = None,
+        filter: Optional[Expression] = None,
+        limit: Optional[int] = None,
+        offset: int = 0,
+):
+    cdef shared_ptr[CDataset] cdataset = dataset.unwrap()
+    cdef shared_ptr[LScannerBuilder] builder = shared_ptr[LScannerBuilder](
+        new LScannerBuilder(cdataset))
+
+    if columns:
+        builder.get().Project([tobytes(c) for c in columns])
+    if filter is not None:
+        builder.get().Filter(_bind(filter, dataset.schema()))
+    if limit is not None:
```

Comment:
Silly edge case but offset is ignored if limit isn't specified. May want to document that

Reply:
SQL standard does not support offset w/o LIMIT IIRC.